src/mpid/ch4/netmod/ofi/ofi_init.c

```diff
@@ -937,23 +948,32 @@ int MPIDI_OFI_mpi_free_mem(void *ptr)
 
 /* ---- static functions for vni contexts ---- */
 static int create_vni_domain(struct fid_domain **p_domain, struct fid_av **p_av,
-                             struct fid_cntr **p_cntr);
+                             struct fid_cntr **p_cntr, int nic);
 static int create_cq(struct fid_domain *domain, struct fid_cq **p_cq);
 static int create_sep_tx(struct fid_ep *ep, int idx, struct fid_ep **p_tx,
-                         struct fid_cq *cq, struct fid_cntr *cntr);
-static int create_sep_rx(struct fid_ep *ep, int idx, struct fid_ep **p_rx, struct fid_cq *cq);
-static int try_open_shared_av(struct fid_domain *domain, struct fid_av **p_av);
-static int create_rma_stx_ctx(struct fid_domain *domain, struct fid_stx **p_rma_stx_ctx);
+                         struct fid_cq *cq, struct fid_cntr *cntr, int nic);
+static int create_sep_rx(struct fid_ep *ep, int idx, struct fid_ep **p_rx, struct fid_cq *cq,
+                         int nic);
+static int try_open_shared_av(struct fid_domain *domain, struct fid_av **p_av, int nic);
 static int open_non_shared_av(struct fid_domain *p_domain, struct fid_av **p_av);
 
-static int create_vni_context(int vni)
+/* This function creates a vni context which includes all of the OFI-level objects needed to
+ * initialize OFI (e.g. domain, address vector, endpoint, etc.). This function takes two arguments:
+ *
+ * vni - The VNI index within a nic to use when assigning the OFI information being created.
+ * nic - The NIC that should be used when setting up the OFI interfaces.
+ *
+ * Each nic will restart its vni indexing. This allows each VNI to use any nic if desired.
```

Comment:
Now sounds like you want vni come before nic, I guess I can get used to that as well as long as it is consistent.

Reply:
In general, I think I put the NIC at the end everywhere to be least disruptive. I don't care strongly where it is.

Reply:
I put the NIC dimension first in the `dest` array because that's where it makes most sense. you need multiple VCI per-"provider" (or NIC), not multiple "providers" per-VCI.

I left the other places in the other order because it was less intrusive and I abstracted away the context index anyway.